google/resource_logging_exclusion.go

```diff
@@ -0,0 +1,258 @@
+	}
+}
+
+func ResourceLoggingExclusionDelete(newUpdaterFunc newResourceLoggingExclusionUpdaterFunc) schema.DeleteFunc {
+	return func(d *schema.ResourceData, meta interface{}) error {
+		config := meta.(*Config)
+		updater, err := newUpdaterFunc(d, config)
+		if err != nil {
+			return err
+		}
+
+		err = updater.DeleteLoggingExclusion(d.Id())
+		if err != nil {
+			return err
+		}
+
+		d.SetId("")
+		return nil
+	}
+}
+
+func expandResourceLoggingExclusion(d *schema.ResourceData, resourceType, resourceId string) (LoggingExclusionId, *logging.LogExclusion) {
+	id := LoggingExclusionId{
+		resourceType: resourceType,
+		resourceId:   resourceId,
+		name:         d.Get("name").(string),
+	}
+
+	exclusion := logging.LogExclusion{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Filter:      d.Get("filter").(string),
+		Disabled:    d.Get("disabled").(bool),
+	}
+	return id, &exclusion
+}
+
+func flattenResourceLoggingExclusion(d *schema.ResourceData, exclusion *logging.LogExclusion) {
+	d.Set("name", exclusion.Name)
+	d.Set("description", exclusion.Description)
+	d.Set("filter", exclusion.Filter)
+	d.Set("disabled", exclusion.Disabled)
+}
+
+func expandResourceLoggingExclusionForUpdate(d *schema.ResourceData) (*logging.LogExclusion, string) {
+	// Can update description/filter/disabled right now.
+	exclusion := logging.LogExclusion{}
+
+	var updateMaskArr []string
+
+	if d.HasChange("description") {
+		exclusion.Description = d.Get("description").(string)
+		exclusion.ForceSendFields = append(exclusion.ForceSendFields, "Description")
+		updateMaskArr = append(updateMaskArr, "description")
+	}
+
+	if d.HasChange("filter") {
+		exclusion.Filter = d.Get("filter").(string)
+		exclusion.ForceSendFields = append(exclusion.ForceSendFields, "Filter")
+		updateMaskArr = append(updateMaskArr, "filter")
+	}
+
+	if d.HasChange("disabled") {
+		exclusion.Disabled = d.Get("disabled").(bool)
+		exclusion.ForceSendFields = append(exclusion.ForceSendFields, "Disabled")
+		updateMaskArr = append(updateMaskArr, "disabled")
+	}
+
+	updateMask := strings.Join(updateMaskArr, ",")
+	return &exclusion, updateMask
+}
+
+// The ResourceLoggingExclusionUpdater interface is implemented for each GCP
+// resource supporting log exclusions.
+//
+// Implementations should keep track of the resource identifier.
+type ResourceLoggingExclusionUpdater interface {
+	CreateLoggingExclusion(parent string, exclusion *logging.LogExclusion) error
+	ReadLoggingExclusion(id string) (*logging.LogExclusion, error)
+	UpdateLoggingExclusion(id string, exclusion *logging.LogExclusion, updateMask string) error
+	DeleteLoggingExclusion(id string) error
+
+	GetResourceType() string
+
+	// Returns the unique resource identifier.
+	GetResourceId() string
+
+	// Textual description of this resource to be used in error message.
+	// The description should include the unique resource identifier.
+	DescribeResource() string
```

Comment:
This was more or less copied from IAM. Not sure if it makes entirely sense like that.